src/main/scala/tech/cryptonomic/conseil/util/DatabaseUtil.scala

```diff
@@ -2,64 +2,154 @@ package tech.cryptonomic.conseil.util
 
 import slick.jdbc.PostgresProfile.api._
 import slick.jdbc.{GetResult, PositionedParameters, SQLActionBuilder}
-import tech.cryptonomic.conseil.generic.chain.DataTypes.Predicate
-import tech.cryptonomic.conseil.tezos.TezosDatabaseOperations.makePredicates
+import tech.cryptonomic.conseil.generic.chain.DataTypes.OperationType.OperationType
+import tech.cryptonomic.conseil.generic.chain.DataTypes.{OperationType, Predicate, QueryOrdering}
 
 /**
   * Utility functions and members for common database operations.
   */
 object DatabaseUtil {
   lazy val db = Database.forConfig("conseildb")
 
-  /** Concatenates SQLActionsBuilders
-    * Slick does not support easy concatenation of actions so we need this function based on https://github.com/slick/slick/issues/1161
-    * @param acc      base SqlActionBuilder with which we want concatenate other actions
-    * @param actions  list of actions to concatenate
-    * @return         one SQLActionBuilder containing concatenated actions
+  /**
+    * Utility object for generic query composition with SQL interpolation
     */
-  def concatenateSqlActions(acc: SQLActionBuilder, actions: List[SQLActionBuilder]): SQLActionBuilder = {
-    actions.foldLeft(acc) {
-      case (accumulator, action) =>
-        SQLActionBuilder(accumulator.queryParts ++ action.queryParts, (p: Unit, pp: PositionedParameters) => {
-          accumulator.unitPConv.apply(p, pp)
-          action.unitPConv.apply(p, pp)
-        })
+  object QueryBuilder {
+    /** Concatenates SQLActionsBuilders
+      * Slick does not support easy concatenation of actions so we need this function based on https://github.com/slick/slick/issues/1161
+      *
+      * @param acc     base SqlActionBuilder with which we want concatenate other actions
+      * @param actions list of actions to concatenate
+      * @return one SQLActionBuilder containing concatenated actions
+      */
+    def concatenateSqlActions(acc: SQLActionBuilder, actions: List[SQLActionBuilder]): SQLActionBuilder = {
```

Comment:
I have a feeling that multiple `actions` are never passed in, so we have to uselessly wrap every single builder passed to this method in a `List(_)` constructor...

An alternative to keep it possible to pass multiple builders would be to make it a vararg
`def contatenateSqlActions(first: SQLActionBuilder, actions: SQLActionBuilder*): SQLActionBuilder = ...`

The implementation won't actually change but calling the method would be smoother

Reply:
you are right, thats a good idea, changed it